subsys/net/lib/lwm2m/lwm2m_observation.c

```diff
@@ -529,7 +529,7 @@ struct observe_node *engine_observe_node_discover(sys_slist_t *observe_node_list
 						  const uint8_t *token, uint8_t tkl)
 {
 	struct observe_node *obs;
-	int obs_list_size, path_list_size;
+	int obs_list_size, path_list_size = 0;
```

Comment:
Why is this change?
On what compiler did you see the error/warning code?

Reply:
running twister using SDK version 0.16.3-rc1 on my local machine. I'm afraid I've lost the precise tree at this point. Looking at the code, I wasn't surprised by the compiler generating a spurious warning here as the assignment and use are both protected inside 'if' clauses. I'd be fine removing this patch, but as I couldn't get a clean twister run locally without it, I included it here assuming it would also be needed for the CI run against the PR.